src/server/rdb_load.cc

```diff
@@ -1487,25 +1508,47 @@ auto RdbLoaderBase::ReadLzf() -> io::Result<LzfString> {
 
 auto RdbLoaderBase::ReadSet(int rdbtype) -> io::Result<OpaqueObj> {
   size_t len;
-  SET_OR_UNEXPECT(LoadLen(NULL), len);
+  if (pending_read_.remaining > 0) {
+    len = pending_read_.remaining;
+  } else {
+    SET_OR_UNEXPECT(LoadLen(NULL), len);
+  }
 
   if (rdbtype == RDB_TYPE_SET_WITH_EXPIRY) {
     len *= 2;
   }
 
+  /*
+   * TODO
+   *
+   * I'm quite confused by the existing logic here, were we limit
```

Comment:
If i remember correctly, it was more efficient to allocate smaller arrays than a single huge array.
And I agree with you - lets start with a single "static" rule of upto kMaxBlobLen for streaming.

Reply:
ok sure